core/node/node_framework/src/implementations/layers/node_storage_init/external_node_strategy.rs

```diff
@@ -73,14 +74,21 @@ impl WiringLayer for ExternalNodeInitStrategyLayer {
             client: client.clone(),
             pool: pool.clone(),
         });
-        let snapshot_recovery = self.snapshot_recovery_config.map(|recovery_config| {
-            Arc::new(ExternalNodeSnapshotRecovery {
-                client: client.clone(),
-                pool: pool.clone(),
-                recovery_config,
-                app_health,
-            }) as Arc<dyn InitializeStorage>
-        });
+        let snapshot_recovery = match self.snapshot_recovery_config {
+            Some(recovery_config) => {
+                let recovery_pool = input
+                    .master_pool
+                    .get_custom(self.max_postgres_concurrency.get() as u32)
+                    .await?;
+                Some(Arc::new(ExternalNodeSnapshotRecovery {
+                    client: client.clone(),
+                    pool: recovery_pool,
+                    recovery_config,
+                    app_health,
+                }) as Arc<dyn InitializeStorage>)
```

Comment:
Bikeshedding: I'd extract `Arc::new(_)` to a local var since this doesn't look very readable.